src/wallet/api/wallet.cpp

```diff
@@ -513,22 +515,29 @@ string WalletImpl::keysFilename() const
     return m_wallet->get_keys_file();
 }
 
-bool WalletImpl::init(const std::string &daemon_address, uint64_t upper_transaction_size_limit)
+void WalletImpl::init(const std::string &daemon_address, uint64_t upper_transaction_size_limit)
 {
     clearStatus();
-    doInit(daemon_address, upper_transaction_size_limit);
-    bool result = this->refresh();
-    // enabling background refresh thread
-    startRefresh();
-    return result;
 
-}
+    m_wallet->init(daemon_address, upper_transaction_size_limit);
 
-void WalletImpl::initAsync(const string &daemon_address, uint64_t upper_transaction_size_limit)
-{
-    clearStatus();
-    doInit(daemon_address, upper_transaction_size_limit);
-    startRefresh();
+    // in case new wallet, this will force fast-refresh (pulling hashes instead of blocks)
+    // If daemon isn't synced a calculated block height will be used instead
+    if (isNewWallet() && daemonSynced()) {
+        LOG_PRINT_L2(__FUNCTION__ << ":New Wallet - fast refresh until " << daemonBlockChainHeight());
```

Comment:
I'd rather start using the new macros if you can, though this is not a request to do that right now, just a general comment, since they have an explicit "severity".